src/skydb-direct.ts

```diff
@@ -0,0 +1,73 @@
+import { pki } from "node-forge";
+import { SkynetClient } from "./client";
+import { RegistryEntry, SignedRegistryEntry } from "./registry";
+import { trimUriPrefix, uriSkynetPrefix } from "./utils";
+
+/**
+ * Gets the JSON object corresponding to the publicKey and dataKey.
+ * @param publicKey - The user public key.
+ * @param dataKey - The key of the data to fetch for the given user.
+ * @param [customOptions={}] - Additional settings that can optionally be set.
+ * @param [customOptions.timeout=5000] - Timeout in ms for the registry lookup.
+ */
+export async function getRegistry(
+  this: SkynetClient,
+  publicKey: string,
+  dataKey: string,
+  customOptions = {}
+): Promise<{ data: string; revision: number } | null> {
+  const opts = {
+    ...this.customOptions,
+    ...customOptions,
+  };
+
+  // lookup the registry entry
+  const entry: SignedRegistryEntry = await this.registry.getEntry(publicKey, dataKey, opts);
+  if (entry === null) {
+    return null;
+  }
+
+  return { data: entry.entry.data, revision: entry.entry.revision };
+}
+
+export async function setRegistry(
+  this: SkynetClient,
+  privateKey: string,
+  dataKey: string,
+  skylink: string,
+  revision?: number,
+  customOptions = {}
+): Promise<void> {
+  const opts = {
+    ...this.customOptions,
+    ...customOptions,
+  };
+
+  const privateKeyBuffer = Buffer.from(privateKey, "hex");
+
+  // Upload the data to acquire its skylink
+  // TODO: Replace with upload request method.
+
+  if (!revision) {
```

Comment:
@kwypchlo because I need this if statement and was easier to copy & delete some code